scenes/game_elements/props/filling_barrel/components/filling_barrel.gd

```diff
@@ -74,3 +82,31 @@ func fill() -> void:
 func _disable_collisions() -> void:
 	hit_box.process_mode = Node.PROCESS_MODE_DISABLED
 	collision_shape_2d.disabled = true
+
+
+func _total_frames() -> int:
+	if not _are_sprite_frames_valid():
+		return 0
+
+	return animated_sprite_2d.sprite_frames.get_frame_count(FILLING_NAME_ANIMATION)
+
+
+func _get_configuration_warnings() -> PackedStringArray:
+	return _sprite_frames_errors()
+
+
+func _sprite_frames_errors() -> PackedStringArray:
+	if not sprite_frames:
+		return ["sprite_frames must be set."]
```

Comment:
This is not needed because the setter ensures that it will be `DEFAULT_SPRITE_FRAMES`.